### PR TITLE
optimize soeprotocol

### PR DESCRIPTION
**Pull request by QuentinGruber**
https://github.com/H1emu/h1emu-core/pull/29 will be a huge improvement also

### CODE REVIEW

**Comment by QuentinGruber** [comment]
Will in game test this then merge

**Comment by QuentinGruber** [comment]
https://github.com/H1emu/h1emu-core/pull/29 will come later on since it less risky than this one

**Comment by QuentinGruber** [comment]
> Will in game test this then merge

Doesn't work as expected accelerated packets and resend in general create to much dupes

**Review by avcio9 — APPROVED** [review]

File: src/servers/SoeServer/soeserver.ts [hunk 1]
```diff
@@ -238,7 +225,19 @@ export class SOEServer extends EventEmitter {
...
+      client.sendingTimer = setTimeout(() => {
+        client.sendingTimer = null;
+        this.sendingProcess(client);
+      }, this._waitTimeMs + additonalTime);
+    }
+  }
+
+  // Handle the packet received from the client
   private handlePacket(client: SOEClient, packet: any) {
+    // console.log("recieved : ", packet.name);
```
Comment (QuentinGruber) [inline src/servers/SoeServer/soeserver.ts:240]:
```suggestion
```

File: src/servers/SoeServer/soeserver.ts [hunk 2]
```diff
@@ -424,6 +394,11 @@ export class SOEServer extends EventEmitter {
               console.error("parsing error " + parsed_data.error);
               console.error(parsed_data);
             } else {
+              if (client.lastKeepAliveTimer) {
+                client.lastKeepAliveTimer.refresh();
+              }
+              // TODO: necessary?
```
Comment (QuentinGruber) [inline src/servers/SoeServer/soeserver.ts:400]: nope so rm this

File: src/servers/SoeServer/soeserver.ts [hunk 3]
```diff
@@ -501,6 +490,121 @@ export class SOEServer extends EventEmitter {
...
+      client,
+      client.waitingQueue
+    );
+    if (waitingQueuePacket) {
+      this._sendAndBuildPhysicalPacket(client, waitingQueuePacket);
+    }
+
+    if (client.unAckData.size > 0) {
+      // console.log("unack data size", client.unAckData.size);
+      // console.log("next sending scheduled in", 100 * client.sendingVsHandleRate);
```
Comment (QuentinGruber) [inline src/servers/SoeServer/soeserver.ts:604]: rm

File: src/servers/ZoneServer2016/zoneserver.ts [hunk 4]
```diff
@@ -7601,18 +7629,16 @@ export class ZoneServer2016 extends EventEmitter {
       }
     }
   }
+  clientRoutineLoop() {}
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/zoneserver.ts:7632]: rm

**Pull request merged**